File: Sudoku/[VB]-Sudoku/VB/sudoku/core/Game.vb
Public Class Game

    Public Event ShowClues(ByVal grid()() As Integer)
    Public Event ShowSolution(ByVal grid()() As Integer)

    Private HRow(8) As List(Of Integer)
    Private VRow(8) As List(Of Integer)
    Private ThreeSquare(8) As List(Of Integer)
    Private grid(8)() As Integer

    Private r As Random

    Public Sub NewGame(ByVal rn As Random)
        Me.r = rn
        createNewGame()
    End Sub

    Private Sub initializeLists()
        For x As Integer = 0 To 8
            HRow(x) = New List(Of Integer)(New Integer() {1, 2, 3, 4, 5, 6, 7, 8, 9})
            VRow(x) = New List(Of Integer)(New Integer() {1, 2, 3, 4, 5, 6, 7, 8, 9})
            ThreeSquare(x) = New List(Of Integer)(New Integer() {1, 2, 3, 4, 5, 6, 7, 8, 9})
            Dim row(8) As Integer
            grid(x) = row
        Next
    End Sub

    Private Sub createNewGame()
        Do
            initializeLists()
            For y As Integer = 0 To 8
                For x As Integer = 0 To 8
                    grid(y)(x) = Nothing
                    Dim si As Integer = (y \ 3) * 3 + (x \ 3)
                    Dim useful() As Integer = HRow(y).Intersect(VRow(x)).Intersect(ThreeSquare(si)).ToArray
                    If useful.Count = 0 Then Continue Do
                    Dim randomNumber As Integer = useful(r.Next(0, useful.Count))
                    HRow(y).Remove(randomNumber)
                    VRow(x).Remove(randomNumber)
                    ThreeSquare(si).Remove(randomNumber)
                    grid(y)(x) = randomNumber
                    If y = 8 And x = 8 Then Exit Do
                Next
            Next
        Loop

        RaiseEvent ShowClues(grid)

    End Sub

    Public Sub showGridSolution()

        RaiseEvent ShowSolution(grid)

    End Sub

End Class

File: Sudoku/[C#]-Sudoku/C#/sudoku cs/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace sudoku_cs
{
    public class Game
    {

        public event ShowCluesEventHandler ShowClues;
        public delegate void ShowCluesEventHandler(int[][] grid);
        public event ShowSolutionEventHandler ShowSolution;
        public delegate void ShowSolutionEventHandler(int[][] grid);

        private List<int>[] HRow = new List<int>[9];
        private List<int>[] VRow = new List<int>[9];
        private List<int>[] ThreeSquare = new List<int>[9];

        private int[][] grid = new int[9][];

        private Random r;
        public void NewGame(Random rn)
        {
            this.r = rn;
            createNewGame();
        }

        private void initializeLists()
        {
            for (int x = 0; x <= 8; x++)
            {
                HRow[x] = new List<int>(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 });
                VRow[x] = new List<int>(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 });
                ThreeSquare[x] = new List<int>(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 });
                int[] row = new int[9];
                grid[x] = row;
            }
        }

        private void createNewGame()
        {
            while (true)
            {
            break1:
                initializeLists();
                for (int y = 0; y <= 8; y++)
                {
                    for (int x = 0; x <= 8; x++)
                    {
                        int si = (y / 3) * 3 + (x / 3);
                        int[] useful = HRow[y].Intersect(VRow[x]).Intersect(ThreeSquare[si]).ToArray();
                        if (useful.Count() == 0)
                        {
                            goto break1;
                        }
                        int randomNumber = useful[this.r.Next(0, useful.Count())];
                        HRow[y].Remove(randomNumber);
                        VRow[x].Remove(randomNumber);
                        ThreeSquare[si].Remove(randomNumber);
                        grid[y][x] = randomNumber;
                        if (y == 8 & x == 8)
                        {
                            goto break2;
                        }
                    }                    
                }                                
            };
            break2:

            if (ShowClues != null)
            {
                ShowClues(grid);
            }

        }


        public void showGridSolution()
        {
            if (ShowSolution != null)
            {
                ShowSolution(grid);
            }

        }

    }
}
